frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/delete-block-pool-modal.tsx

```diff
@@ -86,7 +87,10 @@ const DeleteBlockPoolModal = withHandlePromise((props: DeleteBlockPoolModalProps
 
   // Delete block pool
   const deletePool = () => {
-    handlePromise(k8sKill(CephBlockPoolModel, blockPoolConfig), () => close());
+    handlePromise(k8sKill(CephBlockPoolModel, blockPoolConfig), () => {
+      close();
+      history.goBack();
```

Comment:
We are using the same code for deletion from the list page and details page. history.goback() will affect when deletion is happening from blockPoolListPage

Reply:
Once the pool is created it will redirect to details, history.goBack() will redirect to poolCreate page instead of list page.

Reply:
Here instead of history.goBack(); Check the URL if you are on the Blackpool list page, if you are already on the pool list page then no need to do anything. If else(means you are in details page) redirect to BlockPool list page.

Reply:
const location = useLocation();
```
if location.pathname.includes(poolName) {
    history.push(location.pathname.split(`/${poolName}`)[0]);
}
```

Reply:
Please test deletion from 
  Using 4.9 UI under storage -> openshift-storage-foundation -> StorageSystems -> {storagesystem-name} -> BlockPools
   1. Delete BlockPool from details page.
   2. Delete BlockPool from list page.
   3. Delete BlockPool from details page immediate after pool creation from pool creation UI.
   Using 4.8 UI operators -> installed operators ->  {OCS operator page} -> BlockPool
   1. Delete BlockPool from details page.
   2. Delete BlockPool from list page.
   3. Delete BlockPool from details page immediate after pool creation from pool creation UI.

Reply:
Verified!